core/templates/domain/skill/skill-creation-backend-api.service.ts

```diff
@@ -29,46 +30,75 @@ export interface ISkillCreationBackend {
   description: string,
   'explanation_dict': string,
   'linked_topic_ids': string[],
-  rubrics: IRubricBackend
+  rubrics: IRubricBackend,
+  filenames: string[]
+}
+
+export interface IImageData {
+  filename: string,
+  imageBlob: Blob
 }
 
 @Injectable({
   providedIn: 'root'
 })
 export class SkillCreationBackendApiService {
-  constructor(private http: HttpClient) {}
+  constructor(private csrfTokenService: CsrfTokenService) {}
 
   _createSkill(
       successCallback: (value?: Object | PromiseLike<Object>) => void,
       errorCallback:(reason?: any) => void,
       description: string, rubrics: IRubricBackend, explanation: string,
-      linkedTopicIds: string[]): void {
+      linkedTopicIds: string[], imagesData: IImageData[]): void {
```

Comment:
I think you need a JSDoc for this function that, in particular, has a detailed description of what imagesData (and other args) represent.

Reply:
Added.